Initial usage of the `typescript-sdk`

andreaTP opened the pull request:
Based on #4338 

I updated everything on the Kiota side and converted one simple endpoint to use the TS SDK in the UI.

- should we move everything to Kiota?
- should we do it in a single PR? (there are 45 endpoints, it's doable but not nice to review)
- which `model`s should we use as a source of truth?
- what do you think about the approach for authentication?

EricWittmann commented:
>     * should we move everything to Kiota?

Yes.

>     * should we do it in a single PR? (there are 45 endpoints, it's doable but not nice to review)

Let's get this PR merged first, then I'll convert everything else in a single PR.

>     * which `model`s should we use as a source of truth?

I'll delete all the existing models in favor of the Kiota generated ones.

>     * what do you think about the approach for authentication?

LGTM, although it's a bit hard to review `rest.utils.ts` code check warnings/errors.  LOL what editor are you using that is still using TAB characters?!

andreaTP commented:
> LOL what editor are you using that is still using TAB characters?!

I told you I'm bad at formatting code :stuck_out_tongue: it's probably copy pasted.
I get this PR in a final shape and we can convert the remaining endpoints.
Warning: we "might" need some tweaks for files upload/download but let see ...

andreaTP commented on ui/package.json:
```diff
@@ -20,5 +20,8 @@
         "shelljs": "0.8.5"
+    },
+    "dependencies": {
+        "@apicurio/apicurio-registry-client": "file:../typescript-sdk"
```
this uses `npm link` ... not sure if we can improve it anyhow ...

EricWittmann replied:
ok - i'll see what I can do on this

andreaTP commented on ui/ui-app/src/utils/rest.utils.ts:
```diff
@@ -3,9 +3,63 @@ import { ContentTypes } from "@models/contentTypes.model.ts";
+// END - Kiota Machinery
 
 
+const AXIOS = axios.create();
```
Ideally, at the end of the process, the rest of this file should go away.

andreaTP commented on ui/ui-app/src/utils/rest.utils.ts:
```diff
@@ -3,9 +3,63 @@ import { ContentTypes } from "@models/contentTypes.model.ts";
+
+export function createAuthProvider(auth: AuthService): AuthenticationProvider {
+    if (auth.isOidcAuthEnabled()) {
+        return new TokenAuthenticationProvider("Bearer", async () => auth.getToken().then(v => v!));
```
Just to be sure, are we testing those configurations of the UI in CI?

EricWittmann replied:
What configurations?  We're only doing basic testing of the UI in CI - in two places.  We have UI tests in the registry repo, and we have some in the 3scale dev environment as well.

More UI testing is for sure needed - currents tests are mostly smoke tests.

andreaTP replied:
Do we test the UI in an environment where OIDC is enables and Keycloak configured?

EricWittmann replied:
In the 3scale dev environment, yes.  But not in registry CI.

andreaTP commented:
I hope I have addressed all the comments, a couple of additional questions :slightly_smiling_face: :

- are we convinced about committing the generated code?
- is it ok to rebuild the `ts-sdk `in CI all the time?

EricWittmann commented:
> I hope I have addressed all the comments, a couple of additional questions 🙂 :
> 
>     * are we convinced about committing the generated code?

I'm just getting to this now, but for sure we do **not** want to commit the generated code.  And this PR doesn't seem to do that...which is good.

> 
>     * is it ok to rebuild the `ts-sdk `in CI all the time?

andreaTP commented:
> And this PR doesn't seem to do that...which is good.

I was mistaken due to the LOC changed :sweat_smile: .

> We could optimize I guess - only generate if the openapi.json file changes. But I don't think we're doing that for other SDKs, are we?

No, I think it's fine with this proposed setup.

EricWittmann commented:
Closing in favor of https://github.com/Apicurio/apicurio-registry/pull/4806
